cephci/utils/configure.py

```diff
@@ -76,3 +80,180 @@ def setup_ssh_keys(installer, nodes):
                 SSH_COPYID.format(SSH_ID_RSA_PUB, "root", node.hostname),
             ),
         )
+
+
+def create_private_registry_dirs(node):
+    """
+    Create private registry directories for disconnected install
+    """
+    pass
+    cmd = "mkdir -p /opt/registry/{auth,certs,data}"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to create directory for private registry")
+        return False
+    return True
+
+
+def set_private_registry_creds(node, registry_uname, registry_pwd):
+    """
+    Create credentials for accessing the private registry
+    Args:
+        node (ceph): Node to execute cmd
+        registry_uname (str): Username for the private registry
+        registry_pwd (str): Password for the private registry
+
+    Returns:
+
+    """
+    cmd = f"htpasswd -bBc /opt/registry/auth/htpasswd {registry_uname} {registry_pwd}"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to create credentials for accessing the private registry")
+        return False
+    return True
+
+
+def create_self_signed_certificate(node):
+    """Creates self-signed certificate for private registry"""
+    cmd = (
+        f"openssl req -newkey rsa:4096 -nodes -sha256 -keyout /opt/registry/certs/domain.key -x509 -days 365 -out "
+        f'/opt/registry/certs/domain.crt -addext "subjectAltName = DNS:{node.hostname}" '
+        f'-subj "/C=IN/ST=ST/L=XYZ/O=ORG/CN={node.hostname}"'
+    )
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to create a self-signed certificate")
+        return False
+    return True
+
+
+def create_link_to_domain_cert(node):
+    """Creates a link to the domain certificate"""
+    cmd = "ln -s /opt/registry/certs/domain.crt /opt/registry/certs/registry.crt"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to create a link to the domain certificate")
+        return False
+    return True
+
+
+def add_cert_to_trusted_list(node):
+    """Add the certificate to the trusted list on the private registry node"""
+    cmd = "cp /opt/registry/certs/domain.crt /etc/pki/ca-trust/source/anchors/"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to update-ca-trust")
+        return False
+
+    # Update the ca-trust
+    if not update_ca_trust(node):
+        return False
+    return True
+
+
+def update_ca_trust(node):
```

Comment:
`update_ca_trust` & `validate_trusted_list` can be merged with method name `setup_ca_trust`

Reply:
The above methods just performs a validation of certificate creation. So clubbing them to `setup_ca_trust` will give a wrong interpretation of what these methods do.

Reply:
+1